### ISSUE TITLE
Client crash when copy

### ISSUE THREAD

**Issue by Florian-BACHO** [vasild/cpp-ipfs-http-client#13]
```c++
#include <ipfs/client.h>

int main()
{
        ipfs::Client ipfs("localhost", 5001);
        ipfs::Client ipfsCopy(ipfs); // Segmentation fault

        return (0);
}
```

**Comment by linas** [comment]
This is fixed in pull req #21

**Comment by vasild** [comment]
Fixed, thanks for the report @Florian-BACHO and for the fix @linas.

This was violating the [Rule of Five](https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)#Rule_of_Five), I added the remaining methods that were missing.
